packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -277,26 +276,6 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
             "Transaction gas limit too low to enqueue."
         );
 
-        // We need to consume some amount of L1 gas in order to rate limit transactions going into
-        // L2. However, L2 is cheaper than L1 so we only need to burn some small proportion of the
-        // provided L1 gas.
-        uint256 gasToConsume = _gasLimit/L2_GAS_DISCOUNT_DIVISOR;
```

Comment:
cc @ben-chain, I was under the impression the gas burn here was necessary to prevent what are effectively DoS attacks. Why are we able to remove it now?

Reply:
This isn't a good way to prevent DoS attacks as it does nothing to recoup the cost that the sequencer must bear by the transaction getting through. Instead of burning gas an amount of value should be transferred to the sequencer on L1 to cover the cost of batch submission

Reply:
> does nothing to recoup the cost that the sequencer must bear by the transaction getting through
This is true, although I am comfortable with the high degree of asymmetry of the grief attack.  The cost to the sequencer is 32bytes of calldata vs. an SSTORE, even without any gas burn here.

The ratio is high enough that I would still be comfortable with removing the gas burn logic, but after a discussion earlier today, it sounds like we want to leave it in for now IIRC.

Reply:
@ben-chain @smartcontracts may be worth being explicit: Does this mean this change should be reverted? Could you request changes in the PR if so?

Reply:
Aw man. The gas burn is a real pain to deal with and sending messages already takes a massive amount of gas. 

Currently costs about $50 (100 gwei, $3300 ETH, 150k gas) which is far beyond what any user on L2 would be paying. Feels like the ratelimit is already baked into the >100k of storage overhead.

Reply:
So the real problem is there is no way to know the gas consumption of the L1 to L2 message and a single L1 transaction could in theory send many L1 to L2 messages in a single transaction. A way to solve this would be to have the execution fee paid out to the sequencer on L1 but it is impossible to really know that execution fee up front. This means that a way to do this is to limit the amount of gas used by a L1 to L2 message. There could be a free tier which allows for a `gasLimit` up to a certain value and then higher `gasLimit` require a payment to the sequencer. h/t @karlfloersch